web/services/checks.go

```diff
@@ -266,12 +282,35 @@ func (c *checksService) GetSelectedChecksById(id string) (models.SelectedChecks,
 		SelectedChecks: []string{},
 	}
 
-	result := c.db.Where("id", id).First(&selectedChecks)
-	if errors.Is(result.Error, gorm.ErrRecordNotFound) {
-		return selectedChecks, nil
+	catalog, err := c.GetChecksCatalog()
```

Comment:
I find this new method over complicated.
Is not there any sql join operation we could use?
Maybe join the selected checks with catalog and filter by premium

Reply:
Yes, you are right.
Yet the overcomplication comes from the way the models and entities were designed.
That needs to be followed up and maybe refactored in another PR, I agree.

Reply:
let's bring this home first 🚀